pkg/types/validation/installconfig.go

```diff
@@ -229,3 +233,33 @@ func validatePlatform(platform *types.Platform, fldPath *field.Path, openStackVa
 	}
 	return allErrs
 }
+
+func validateProxy(p *types.Proxy, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if p.HTTPProxy == "" && p.HTTPSProxy == "" && p.NoProxy == "" {
```

Comment:
i would have assumed, atleast one of `HTTPProxy` and `HTTPSProxy`; user could have empty no-proxy list.

Reply:
I am updating the conditional to:
```
 p.HTTPProxy == "" && p.HTTPSProxy == "" && p.NoProxy != "*"
```

I believe this is a valid edge case, which effectively disables the proxy.

Reply:
Setting `HTTPProxy` and `HTTPSProxy` to empty are sufficient to disable the proxy. Any valid value in `NoProxy` would be effectively ignored since there is no proxy to bypass. However, the way to disable the proxy is to exclude the `Proxy` field form the install config. If the `Proxy` field is present, then at least on of `HTTPProxy` and `HTTPSProxy` should be present. The user could still choose to disable the proxy by setting `NoProxy` to `*`, but even if this case at least one of `HTTPProxy` and `HTTPSProxy` should be set.

I don't think that the installer should support setting `*` for `NoProxy` at all. The necessity for supporting that in the API is that it is the way to disable the proxy, where a `Proxy` resource should exist. In the installer, however, the preferred way to disable the proxy should be to omit `Proxy` from the install config.

Reply:
> The user could still choose to disable the proxy by setting NoProxy to *, but even if this case at least one of HTTPProxy and HTTPSProxy should be set.

This makes sense. 

> I don't think that the installer should support setting * for NoProxy at all.

I am inclined to agree, but my reservation is that it was supported in 3.11 and I believe the situation was similar (proxy could just be omitted from the Ansible inventory). So it would be good to know if there was an intention behind the 3.11 design.

Reply:
@sdodson do you know the intent ^